src/gradualizer_file_utils.erl

```diff
@@ -0,0 +1,64 @@
+%%% @doc Module with utility functions to work with files.
+
+-module(gradualizer_file_utils).
+
+-export([
+            get_forms_from_erl_safe/1,
+            get_forms_from_beam_safe/1,
+            get_forms_from_erl/1,
+            get_forms_from_beam/1
+        ]).
+
+-type abstract_form() :: [erl_parse:abstract_form() | erl_parse:form_info()].
+
+-type parsed_file_error() :: {file_not_found, file:filename()} |
+                             {file_open_error, {file:posix() | badarg | system_limit, file:filename()}}.
+
+-type parsed_file() :: {ok, abstract_form()} |
+                       parsed_file_error().
+
+-export_type([parsed_file_error/0, abstract_form/0]).
+
+-spec get_forms_from_erl_safe(file:filename()) -> parsed_file().
+get_forms_from_erl_safe(File) ->
+    case epp:parse_file(File, []) of
+        {ok, Forms} ->
+            {ok, Forms};
+        {error, enoent} ->
+            {file_not_found, File};
+        {error, Reason} ->
+            {file_open_error, {Reason, File}}
+    end.
+
+-spec get_forms_from_beam_safe(file:filename()) -> parsed_file().
+get_forms_from_beam_safe(File) ->
+    case beam_lib:chunks(File, [abstract_code]) of
+        {ok, {_Module, [{abstract_code, {raw_abstract_v1, Forms}}]}} ->
+            {ok, Forms};
+        {ok, {_Module, [{abstract_code,no_abstract_code}]}} ->
+            {forms_not_found, File};
+        {error, beam_lib, {file_error, _, enoent}} ->
+            {file_not_found, File};
+        {error, beam_lib, {file_error, _, Reason}} ->
+            {file_open_error, {Reason, File}};
+        {error, beam_lib, Reason} ->
+            {forms_error, Reason}
+    end.
+
+-spec get_forms_from_erl(file:filename()) -> abstract_form() | no_return().
+get_forms_from_erl(File) ->
+    case get_forms_from_erl_safe(File) of
+        {ok, Forms} ->
```

Comment:
Sure we can
In Elixir it's common pattern to have 2 versions of utility function: safe and unsafe, for example

```
Map.fetch(map(), key()) :: {:ok, value()} | :error

Map.fetch!(map(), key()) :: value() | no_return()
```

So first idea was to make `get_forms_from_erl` and `get_forms_from_erl!` functions, but it looks not very good because `!` is keyword in Erlang. I searched some info about safe/unsafe functions naming conventions in Erlang, but it seems there is not any